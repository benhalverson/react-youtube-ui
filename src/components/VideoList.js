import React from 'react';
import VideoItem from './VideoItem';

const VideoList = ({ videos, onVideoSelect }) => {
  // has props.videos

  const renderedList = videos
    // some video data are a "channel", so it can't be rendered as video
    .filter(video => video.id && video.id.videoId)
    .map(video => {
      return (
        <VideoItem
          key={video.id.videoId}
          video={video}
          onVideoSelect={onVideoSelect}
        />
      );
    });
  return <div className="ui relaxed divided list">{renderedList}</div>;
};
export default VideoList;
