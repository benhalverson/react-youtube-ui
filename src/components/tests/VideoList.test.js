import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react'
import VideoList from '../VideoList'
import VideoItem from '../VideoItem'

Enzyme.configure({ adapter: new Adapter() });

describe('Component: VideoList', () => {

  const videos = [
    {
      id: {
        videoId: '123'
      },
      snippet: {
        thumbnails: {
          medium: {
            url: 'imageurl.jpg'
          }
        },
        title: 'Hello world'
      }
    },
    {
      id: {
        videoId: '456'
      },
      snippet: {
        thumbnails: {
          medium: {
            url: 'imageurl2.jpg'
          }
        },
        title: 'Hello again'
      }
    },
    {
      id: {
        videoId: '789'
      },
      snippet: {
        thumbnails: {
          medium: {
            url: 'imageurl3.jpg'
          }
        },
        title: 'Hello you'
      }
    }
  ]
  
  test('should render', () => {
    const wrapper = Enzyme.shallow(<VideoList videos={ videos } />)

    expect(wrapper.render()).toMatchSnapshot();
  });

  test('should filter non-video', () => {
    const modifiedVideos = [
      ...videos,
      {
        id: {
          channelId: '313'
        }
      }
    ]
    const wrapper = Enzyme.shallow(<VideoList videos={ modifiedVideos } />)

    const items = wrapper.find(VideoItem)

    // the added item shouldn't be counted since it is not a video
    expect(items.length).toBe(modifiedVideos.length - 1)
  })

  test('should trigger event on select child', () => {
    const spyEvent = jest.fn()
    const wrapper = Enzyme.shallow(<VideoList videos={ videos } onVideoSelect={ spyEvent } />)    
    const items = wrapper.find(VideoItem)

    items.first().prop('onVideoSelect')()

    expect(spyEvent).toBeCalled()
  })

});