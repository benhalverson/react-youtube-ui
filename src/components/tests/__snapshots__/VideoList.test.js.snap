// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: VideoList should render 1`] = `
"<div className=\\"ui relaxed divided list\\">
  <VideoItem video={{...}} onVideoSelect={[undefined]} />
  <VideoItem video={{...}} onVideoSelect={[undefined]} />
  <VideoItem video={{...}} onVideoSelect={[undefined]} />
</div>"
`;
